In [1]:
import ROOT

Welcome to JupyROOT 6.15/01


In [11]:
import math
import numpy

In [2]:
file = ROOT.TFile("../NanoAOD-DYJetsToLL.root")
tree = file.Get("tree")

In [3]:
tree.SetBranchStatus("*", 0)
tree.SetBranchStatus("nMuon", 1)
tree.SetBranchStatus("Muon_pt", 1)
tree.SetBranchStatus("Muon_eta", 1)

In [12]:
pz = numpy.empty(552056, dtype=numpy.float32)

In [17]:
%%timeit
i = 0
for event in tree:
    for pt, eta in zip(event.Muon_pt, event.Muon_eta):
        pz[i] = pt * math.sinh(eta)
        i += 1

1 loop, best of 3: 45.9 s per loop


53.9 sec the first time, 45.1 the second, 45.9 the third